### PR TITLE
StatusTracker: Fix different case paths not matching

### PR DESCRIPTION
**Pull request by jturcotte**
Use a custom std::map comparator functor to do all comparisons
on contained QStrings using Qt::CaseInsensitive on macOS and Windows.

Issue #5257

### CODE REVIEW

File: src/libsync/syncfilestatustracker.cpp [hunk 1]
```diff
@@ -25,20 +25,49 @@ namespace OCC {
 
 Q_LOGGING_CATEGORY(lcStatusTracker, "sync.statustracker", QtInfoMsg)
 
-static SyncFileStatus::SyncFileStatusTag lookupProblem(const QString &pathToMatch, const std::map<QString, SyncFileStatus::SyncFileStatusTag> &problemMap)
+static int pathCompare( const QString& lhs, const QString& rhs )
+{
+    // Should match Utility::fsCasePreserving, we want don't want to pay for the runtime check on every comparison.
```
Comment (ogoffart) [inline src/libsync/syncfilestatustracker.cpp]: And yet, it does not becuase it does not check the env variable.
I suggest you still use that, is the runtime cost so high? 
If you really want to save the runtime cost on every call, i suggest you pass the case sensitivity as a parametter of the comparator.

File: test/testsyncfilestatustracker.cpp [hunk 2]
```diff
@@ -260,6 +260,24 @@ private slots:
...
+        fakeFolder.serverErrorPaths().append("A/a1");
+        fakeFolder.localModifier().appendByte("A/a1");
+
+        fakeFolder.syncOnce();
+        QCOMPARE(fakeFolder.syncEngine().syncFileStatusTracker().fileStatus(""), SyncFileStatus(SyncFileStatus::StatusWarning));
+        QCOMPARE(fakeFolder.syncEngine().syncFileStatusTracker().fileStatus("A"), SyncFileStatus(SyncFileStatus::StatusWarning));
+        QCOMPARE(fakeFolder.syncEngine().syncFileStatusTracker().fileStatus("A/a1"), SyncFileStatus(SyncFileStatus::StatusError));
+        QCOMPARE(fakeFolder.syncEngine().syncFileStatusTracker().fileStatus("B"), SyncFileStatus(SyncFileStatus::StatusWarning));
+
+        // Should still get the status for different casing on macOS and Windows.
```
Comment (ogoffart) [inline test/testsyncfilestatustracker.cpp]: But do we even run the tests on different platform?

Well, at least we could simulate it with env variable.

File: src/libsync/syncfilestatustracker.cpp (hunk 1 above)
Comment (jturcotte) [reply src/libsync/syncfilestatustracker.cpp]: Arf just saw "approved" in the email and didn't see the comment.
The env variable is just for auto tests anyway, no?

File: test/testsyncfilestatustracker.cpp (hunk 2 above)
Comment (jturcotte) [reply test/testsyncfilestatustracker.cpp]: I run them on macOS, and the CI ran them on Linux, so at least both cases were checked once.

**Pull request merged**